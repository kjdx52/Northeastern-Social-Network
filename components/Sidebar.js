import React from "react";
import Sidebaritem from "./Sidebaritem";
import { ImUsers } from "react-icons/im";
import { MdGroups } from "react-icons/md";
import { AiOutlineUser } from "react-icons/ai";
import Image from "next/image";
import { useSession } from "next-auth/react";

/**
 * Sidebar component, displaying user information and various sidebar items.
 */
const Sidebar = () => {
    const { data: session } = useSession();

    return (
        <div className="hidden lg:inline-flex flex-col py-2 pl-2 max-w-xl lg:min-w-[320px]">
            {/* User information section */}
            {session.user.image ? (
                <div className="flex items-center space-x-2 py-3 pl-4 hover:bg-gray-200 rounded-l-xl cursor-pointer">
                    <Image
                        src={session.user.image}
                        height={40}
                        width={40}
                        className="rounded-full cursor-pointer"
                        alt={"photo_id"}
                    />
                    <p className="hidden sm:inline-flex font-medium">
                        {session?.user.name}
                    </p>
                </div>
            ) : (
                <Sidebaritem Icon={AiOutlineUser} value={session?.user.name} />
            )}

            {/* Sidebar items */}
            <Sidebaritem Icon={ImUsers} value="Friends" />
            <Sidebaritem Icon={MdGroups} value="Groups" />
            {/* Uncomment below lines for additional sidebar items */}
            {/* <Sidebaritem Icon={AiOutlineShop} value="MarketPlace"/>
               <Sidebaritem Icon={MdOutlineOndemandVideo} value="Watch"/>
               <Sidebaritem Icon={BsStopwatch} value="Memeries"/>
               <Sidebaritem Icon={MdOutlineExpandMore} value="See more"/> */}
        </div>
    );
};

export default Sidebar;
