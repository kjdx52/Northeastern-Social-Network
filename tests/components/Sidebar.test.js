import { render, screen } from '@testing-library/react';
import Sidebar from '@/components/Sidebar'
import React from 'react';
import { useSession } from 'next-auth/react';

jest.mock('next-auth/react');


// Test the rendering of the sidebar when the user is logged in.
test('renders Sidebar with user session', () => {
    const mockSession = {
        user: {
            name: 'John Doe',
            image: null
        }
    };
    useSession.mockReturnValue({ data: mockSession, status: 'authenticated' });

    render(<Sidebar />);
    expect(screen.getByText('John Doe')).toBeInTheDocument();

});



// Test the rendering when the user's profile image is not available.
test('renders Sidebar with user name and default icon when image is not available', () => {
    const mockSessionWithoutImage = {
        user: {
            name: 'Jane Doe'
        }
    };
    useSession.mockReturnValue({ data: mockSessionWithoutImage, status: 'authenticated' });

    render(<Sidebar />);
    expect(screen.getByText('Jane Doe')).toBeInTheDocument();
});



// Test if the sidebar items are displayed correctly.
test('displays sidebar items correctly', () => {
    const mockSession = {
        user: {
            name: 'John Doe',
            image: 'https://example.com/john-doe.jpg'
        }
    };
    useSession.mockReturnValue({ data: mockSession, status: 'authenticated' });

    render(<Sidebar />);
    expect(screen.getByText('Friends')).toBeInTheDocument();
    expect(screen.getByText('Groups')).toBeInTheDocument();
    // Test for the presence of other Sidebaritem components
});



// Test if the Sidebar component matches the snapshot for consistency over time.
test('Sidebar component matches snapshot', () => {
    const mockSession = {
        user: {
            name: 'John Doe',
            image: 'https://example.com/john-doe.jpg'
        }
    };
    useSession.mockReturnValue({ data: mockSession, status: 'authenticated' });

    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
});
