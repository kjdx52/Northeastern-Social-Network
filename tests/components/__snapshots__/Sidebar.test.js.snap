// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="hidden lg:inline-flex flex-col py-2 pl-2 max-w-xl lg:min-w-[320px]"
  >
    <div
      class="flex items-center space-x-2 py-3 pl-4 hover:bg-gray-200 rounded-l-xl cursor-pointer"
    >
      <img
        alt="photo_id"
        class="rounded-full cursor-pointer"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=https%3A%2F%2Fexample.com%2Fjohn-doe.jpg&w=96&q=75"
        srcset="/_next/image?url=https%3A%2F%2Fexample.com%2Fjohn-doe.jpg&w=48&q=75 1x, /_next/image?url=https%3A%2F%2Fexample.com%2Fjohn-doe.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <p
        class="hidden sm:inline-flex font-medium"
      >
        John Doe
      </p>
    </div>
    <div
      class="flex items-center space-x-2 py-3 pl-4 hover:bg-gray-200 rounded-l-xl cursor-pointer"
      data-testid="MdGroups"
    >
      <svg
        class="h-8 w-8 text-red-300"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        version="1.1"
        viewBox="0 0 18 16"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M12 12.041v-0.825c1.102-0.621 2-2.168 2-3.716 0-2.485 0-4.5-3-4.5s-3 2.015-3 4.5c0 1.548 0.898 3.095 2 3.716v0.825c-3.392 0.277-6 1.944-6 3.959h14c0-2.015-2.608-3.682-6-3.959z"
        />
        <path
          d="M5.112 12.427c0.864-0.565 1.939-0.994 3.122-1.256-0.235-0.278-0.449-0.588-0.633-0.922-0.475-0.863-0.726-1.813-0.726-2.748 0-1.344 0-2.614 0.478-3.653 0.464-1.008 1.299-1.633 2.488-1.867-0.264-1.195-0.968-1.98-2.841-1.98-3 0-3 2.015-3 4.5 0 1.548 0.898 3.095 2 3.716v0.825c-3.392 0.277-6 1.944-6 3.959h4.359c0.227-0.202 0.478-0.393 0.753-0.573z"
        />
      </svg>
      <p
        class="hidden sm:inline-flex font-medium"
      >
        Friends
      </p>
    </div>
    <div
      class="flex items-center space-x-2 py-3 pl-4 hover:bg-gray-200 rounded-l-xl cursor-pointer"
      data-testid="MdGroups"
    >
      <svg
        class="h-8 w-8 text-red-300"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        stroke-width="0"
        viewBox="0 0 24 24"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M12 12.75c1.63 0 3.07.39 4.24.9 1.08.48 1.76 1.56 1.76 2.73V18H6v-1.61c0-1.18.68-2.26 1.76-2.73 1.17-.52 2.61-.91 4.24-.91zM4 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm1.13 1.1c-.37-.06-.74-.1-1.13-.1-.99 0-1.93.21-2.78.58A2.01 2.01 0 000 16.43V18h4.5v-1.61c0-.83.23-1.61.63-2.29zM20 13c1.1 0 2-.9 2-2s-.9-2-2-2-2 .9-2 2 .9 2 2 2zm4 3.43c0-.81-.48-1.53-1.22-1.85A6.95 6.95 0 0020 14c-.39 0-.76.04-1.13.1.4.68.63 1.46.63 2.29V18H24v-1.57zM12 6c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3z"
        />
      </svg>
      <p
        class="hidden sm:inline-flex font-medium"
      >
        Groups
      </p>
    </div>
  </div>
</DocumentFragment>
`;
